api/src/pcapi/core/offerers/api.py

```diff
@@ -2740,23 +2740,16 @@ def create_offerer_address_from_address_api(
             insee_code=address_info.citycode,
             ban_id=address_info.id,
         )
-    except api_adresse.NoResultException:
-        insee_code = None
-        if city and postal_code:
-            try:
-                insee_code = api_adresse.get_municipality_centroid(city, postal_code).citycode
-            except api_adresse.AdresseException:
-                pass
```

Comment:
Ca me semble risqué d'enlever le try/except malgré tout. Est-ce que l'on peut le garder pour plus de sécurité 🙏  ?

Reply:
Je ne vois pas à quoi servir, si `NoResultException` est raised par le connecteur de l’API adresse, c’est suite à ce bout de code:

```
    def get_single_address_result(
        self,
        address: str,
        postcode: str | None,
        city: str | None = None,
        citycode: str | None = None,
    ) -> AddressInfo:
        """
        No human interaction so we limit to 1 result, and add a filter on the INSEE code (Code Officiel Géographique)
        This will get the highest score result from the query, for a specific INSEE code.
        An incorrect result would still be in the vicinity of the expected result, and can be later edited in pc pro
        see https://forum.etalab.gouv.fr/t/interpretation-du-score/3852/4
        If no result is found, we return the centroid of the municipality
        """
        params = {
            "q": address,
            "postcode": postcode,
            "citycode": citycode,
            "autocomplete": 0,
            "limit": 1,
        }

        data = self._cached_search(params=params)
        if self._is_result_empty(data):
            logger.info(
                "No result from API Adresse for queried address",
                extra={"queried_address": address, "postcode": postcode},
            )
            if city is not None and postcode is not None:
                return self.get_municipality_centroid(city=city, postcode=postcode, citycode=citycode)
            raise NoResultException
```

L'api adresse ne nous aura pas renvoyé de résultat, donc on fallback sur le centroïde de la commune qui ne renvoit pas de résultat non plus. `NoResultException` est raised, que l’on catch ici, pour à nouveau demander le centroïde de la commune, qui va de nouveau raised `NoResultException` ?

Reply:
Vu ensemble, on appel l’API adresse tout de même pour récupérer le code insee. J’ai mis à jour la PR avec deux cas d’usage:

- édition manuelle + appel à l’api pour récupérer le code insee
- édition manuelle + appel à l’api qui raise une `NoResultException` pour s’assurer qu’on ne bloque pas bêtement le parcours de création d’offre à cause de la BAN)